datafusion/expr/src/logical_plan/plan.rs

```diff
@@ -1019,6 +1030,17 @@ pub struct DropTable {
     pub schema: DFSchemaRef,
 }
 
+/// Drops a view.
```

Comment:
Not for this PR, but it seems to me like requiring new `LogicalPlan` nodes for each type of  DDL (`CREATE`, `DROP`, etc) is somewhat cumbersome. Maybe we can eventually refactor type of thing into `LogicalPlan::DDL(DDL)` and keep all the DDL related structs in their own structure. I'll file a follow on ticket for this

Reply:
Filed https://github.com/apache/arrow-datafusion/issues/3349 as a follow on

Reply:
Yes, I was wondering about this and follow on work for dropping schema, etc. A lot of plumbing here, and I'm not sure it works if you have your own CatalogProvider.

Reply:
> Yes, I was wondering about this and follow on work for dropping schema, etc. A lot of plumbing here, and I'm not sure it works if you have your own CatalogProvider.

I agree -- it does not work if you have your own CatalogProvider without some more work. I this behavior is fine, as long as the point is clear to users.